### ISSUE TITLE
 REQ HELP: Range 6 - On/Off for convoy escort

### ISSUE THREAD

**Issue by 132ndNeck** [132nd-vWing/ATRM#52]
In order to have it possible for all aircrafts to use the convoy escort mission, I need some assistance.

Is it possible to adda F10 menu to start the convoy at Range 6?
Also, is it possible to have a F10 option to pause the convoy ? So in case a situation develops, the FL in charge of the mission can pause the convoy, handle the situation before the convoy resume its mission?

Convoy is already in the . miz and have the following name:
R6_Convoy1

**FEATURE 1: :**
I have added 3 more convoys, called :

- R6_Convoy2
- R6_Convoy3
- R6_Convoy4

All currently set as late activation.
Would it be possible to make the F10 option so that when selecting a convoy, then one of the convoys (1-4 ) get activated, and use the information above?

Currently 
R6_Convoy1 is set to start with KA-50s are in the vicinity, but I want to remove this if we can get the F10 option, so others can also use it.

And with the 4 convoys options it is easier for randomness and various action taking place.

**FEATURE 2: SAM Threats**
There are 3 SAMs placed at the area. They are named:

- R6_Convoy_SAM1
- R6_Convoy_SAM2
- R6_Convoy_SAM3

They are set as late activation.
Is it possible when the F10 option to activate the convoy, then there is a 50% chance for one of these SAMs to be activated?

**FEATURE 3: Two convoys**
In case there are multiple flights together, in an advanced training. Is it possible to have a option for activating a second convoy (Pick one of the remaining that was not activated for the first convoy).

**Comment by 132nd-Entropy** [comment]
done. Sam will spawn in either case, but with AI off. Once the convoy starts moving, there will be a 50% chance that the SAM will turn on its AI. Also, when the 1st convoy is moving, a second convoy option will appear to also control a s second convoy.

**Comment by 132ndNeck** [comment]
@132nd-Entropy 
Just got back to this after seeing the issues Limbo had last wednesday.

I tested now, and it seems the spawn changes the group name:
![bilde](https://user-images.githubusercontent.com/26002570/75582616-4b07ed80-5a6c-11ea-9bba-75f3ac5e6ece.png)


This messes up the enemy activity, as they are depending on the group in the .miz spawning (with the name set in the .miz). As all the triggers that create activity is based on the group (with correct name) driving through a zone.
So, when the group is given a new name, as shown above, then it will not work. Is it possible to set the group to spawn with the name set in the .miz in the ME?


Same group as above, from the ME:
![bilde](https://user-images.githubusercontent.com/26002570/75582812-b2be3880-5a6c-11ea-9485-2d8545178f71.png)

I assume the same will be valid for all the various convoys.

**Comment by 132nd-Entropy** [comment]
easiest is to do it completely via scripting, or do it via the ME but mixing both will result in issues like this. I can either write a script that replaces your triggers in the ME, or we revert to the old method of spawning the stuff, but then you cant have the menus organized properly etc

**Comment by 132ndNeck** [comment]
Copy, I am happy with completely with scripting as long as you are happy with it :)

(I was just hoping to make it easier for you with the combination, as you then did not need to do too much with it and just needed activate the right unit via F10 menu, and everything else was already setup and working).

**Comment by 132nd-Entropy** [comment]
there is no way around spawning via script and not having a #001, #002 etc added behind the name of the late-activated template. there are a lot of triggers now in there and its hard to follow up on this. I Think the easiest way to fix this is simply do the spawning via triggers in the ME completely. and then use flags to activate the menus. that way we get the simplicity of the ME with the added benefit of properly nesting the menu system via script. 

in the future if we keep in mind that script-spawned units cannot be used by name, as they get renamed and there is no way around it, we should be good.

**Comment by 132nd-Entropy** [comment]
closing for now
